# Data Science

With the support of supplemental funding, universities hosting  National Library of Medicine [Biomedical Informatics and Data Science Research Training Programs](https://www.nlm.nih.gov/ep/GrantTrainInstitute.html) developed data science training materials. 

As one of the goals of this initiatiave was to providing these materials as shareable resources for the educational community, links to these resources are provided below, grouped by institution.

---
# [University of Pittsburgh](https://www.dbmi.pitt.edu)

*Note*: These resources can be also be found at [datascience.dbmi.pitt.edu](https://datascience.dbmi.pitt.edu) 

## Social Media Data Science

[![DOI](https://zenodo.org/badge/147564959.svg)](https://zenodo.org/badge/latestdoi/147564959)

[GitHub Repository](https://github.com/dbmi-pitt/SocialMediaDataScience)

Social media can be a useful source of information regarding public discussion of health issues.  Using this data requires informatics strategies for retrieving, storing, annotating, interpreting, and classifying this data. 

This module challenges users to explore Twitter data, using the Python programming language, Twitter application programming interfaces, and libraries for natural language processing and machine learning.  Students will gain familiarity with the use of these tools to retrieve data, annotate data, conduct basic natural language processing tasks,  and build and evaluate simple classifier.

Students should have some background in Python programming. A Twitter developer account is required. Instructions on acquiring and configuring Twitter accounts are included.

# Differential expression analysis of RNA-Seq data from TCGA

[![DOI](https://zenodo.org/badge/170748515.svg)](https://zenodo.org/badge/latestdoi/170748515)

[GitHub Repository](https://github.com/dbmi-pitt/tcga_brca_rnaseq)

The examination of gene expression patterns in breast cancer samples can provide valuable insights into genetic activity underlying cancers and distinguishing between cancer subtypes. Although the public availability of gene expression data sets through resources such as the Cancer Genome Atlas (TCGA) provides opportunities for exploring and analyzing these datasets, asking meaningful questions often requires the examination of multiple data sets from various sources, manipulation of the data in preparation for analysis, and use of research tools for conducting analyses.

This project explores differential analysis of RNA-Seq data from TCGA, using the R statistical environment and Bioconductor tools, primarily the DESeq2 differential gene analysis package.

# Lumping and splitting: Bias, Variance, Smoothing, and Shrinking

[![DOI](https://zenodo.org/badge/164016584.svg)](https://zenodo.org/badge/latestdoi/164016584)

[GitHub Repository](https://github.com/dbmi-pitt/Bias-variance-smoothing-shrinking)

[Shiny Demo](https://dbmipittdatascience.shinyapps.io/Bias-variance-smoothing-shrinking/)

An explanation of bias and variance and related implications for precision medicine, including consequences of using different definitions of groups for examinations of treatment effects.

# Microbiome Analysis 

[![DOI](https://zenodo.org/badge/164003850.svg)](https://zenodo.org/badge/latestdoi/164003850)

[GitHub Repository](https://github.com/dbmi-pitt/MicrobiomeModule) 

An introduction to the  use of [DECIPHER](http://www2.decipher.codes) tools to analyze  microbiome data, suitable for an introductory programming course. Topics covered include reading data, exploring the abundance of sequences for each sample, examining rank abundance curves, identifying abundant taxa, examining both alpha- and beta-diversity, identifying OTUS, and conducting principal component analysis to compare samples.

# Image Segmentation and Slurm Job Processing

[![DOI](https://zenodo.org/badge/164016740.svg)](https://zenodo.org/badge/latestdoi/164016740)

[GitHub Repository](https://github.com/dbmi-pitt/ImageSegmentation-and-Slurm)

Two modules exploring key data science topics:

* Image Segmentation through Deep Learning.
* Using R and Slurm for high-performance computing

---
# Rice University 

[Website with course materials](http://dstoolsandmodels.rice.edu/)

This site contains source material for the semester-long course "Tools and Models for Data Science", 
developed with funding from the NLM.  "Tools and Models for Data Science"
provides a very wide and yet reasonably deep view of the field of data science.
As the name implies, much of the course is concerned with the practice
and theory of modern tools (software systems) for data science.
Relational databases and SQL are covered in depth, as well as
MapReduce and Spark. The course covers the basics of mathematical
programming with NumPy—especially the importance of vectorized
programming—and deep learning systems such as TensorFlow. In
addition to those systems, the course covers basic statistical
modeling, basic and intermediate topics in optimization (optimization
is the foundation of “learning” in machine learning systems) and the
foundations of machine learning.
=======
# [Yale University](https://medicine.yale.edu/ycmi/)

# Introduction to B cell repertoire analysis with the Immcantation framework

[Bitbucket Repository](https://bitbucket.org/kleinstein/immcantation/src/default/training/)

An introductory module to the use of the [Immcantation framework](http://immcantation.org) to analyze high-throughput adaptive immune receptor repertoire sequencing (AIRR-seq) data. This tutorial covers common investigative approaches and pitfalls in AIRR-seq data analysis, including:

<a href="http://immcantation.org"><img src="https://bitbucket.org/kleinstein/immcantation/raw/5357d31d42e37e55671e4fa10a8b9787dc69a45f/docs/_static/logo.svg" width="18%" align="right"></a>

* V(D)J gene annotation and novel polymorphism detection
* Clonotype assignment
* Diversity analysis
* Mutational load profiling
* Modeling of somatic hypermutation targeting
* Quantification of selection pressure
>>>>>>> b0e7a9044ce7bf5946a684bb5881fc51eb35b80e
